### PR TITLE
Migrate deprecated `rollback()` call

### PR DESCRIPTION
**Pull request by findinpath**
Related tests to test the functionality for regressions:

- `io.trino.plugin.iceberg.TestIcebergOrcConnectorTest#testRollbackSnapshot`
- `io.trino.plugin.iceberg.TestIcebergParquetConnectorTest#testRollbackSnapshot`

### CODE REVIEW

**Review by losipiuk — APPROVED** [review]

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/RollbackToSnapshotProcedure.java [hunk 1]
```diff
@@ -65,6 +65,6 @@ public void rollbackToSnapshot(ConnectorSession clientSession, String schema, St
     {
         SchemaTableName schemaTableName = new SchemaTableName(schema, table);
         Table icebergTable = catalogFactory.create().loadTable(clientSession, schemaTableName);
-        icebergTable.rollback().toSnapshotId(snapshotId).commit();
+        icebergTable.manageSnapshots().rollbackTo(snapshotId).commit();
```
Comment (losipiuk) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/RollbackToSnapshotProcedure.java:68]: Javadoc suggests `setCurrentSnapshot` should be used. Is there a difference?

Comment (findinpath) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/RollbackToSnapshotProcedure.java:68]: You are right, `org.apache.iceberg.RollbackToSnapshot#toSnapshotId` does use `setCurrentSnapshot(long)` as well.

Sorry for the confusion.

**Review by losipiuk — APPROVED** [review]

**Review by alexjo2144 — APPROVED** [review]

**Comment by alexjo2144** [comment]
@findinpath can you resolve the conflicts?

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**